### PR TITLE
libtorrent-rasterbar-1_2_x: 1.2.11 -> 1.2.19

### PR DESCRIPTION
**Pull request by sergv**
I noticed that `tribler` wants `libtorrent==1.2.19` (https://github.com/Tribler/tribler/blob/main/requirements.txt) so decided to update even though it seems to work with the current version.

Changelog: https://github.com/arvidn/libtorrent/blob/v1.2.19/ChangeLog.

The build instructions are since the library now depends on boost-build and still uses AFAIU mostly deprecated as of now Python’s `setup.py`. It's because of `setup.py` I've added `PYTHON_INSTALL_PARAMS` so that it will install python module in a way that Python 3.10+ can find and use. I don't really know what I'm doing here but if I don't adjust  `PYTHON_INSTALL_PARAMS` then `setup.py` will install module as a Python egg which Python 3.10 cannot load. If someone more knowledgeable about Python packaging can review this that would be much appreciated.

The `setup.py` will call Boost's `b2` build system and by default it will look for static `libboost_system`, 'libboos_python3.11`, `libcrypto`, and `libssl`. That can be changed to shared but on my machine the resulting Python `.so` module gets built with undefined symbols and cannot be loaded so static default seems like the best option.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sergv** [comment]
@xvapx, @mkg20001 as nixpkgs maintainers of `tribler`, if you please could review update of `libtorrent-rasterbar` used by `tribler` it would be very useful, thanks.

**Comment by sergv** [comment]
Bump please

File: pkgs/by-name/li/libtorrent-rasterbar-1_2_x/package.nix [hunk 1]
```diff
@@ -17,17 +30,21 @@ in stdenv.mkDerivation {
...
-  nativeBuildInputs = [ automake autoconf libtool pkg-config ];
+  nativeBuildInputs = [ automake autoconf boost-build libtool pkg-config python311.pkgs.setuptools ];
 
-  buildInputs = [ boostPython openssl zlib python311 libiconv ncurses ]
+  buildInputs = [ boostPython opensslStatic zlib python311 libiconv ncurses ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
 
-  preConfigure = "./autotool.sh";
+  preConfigure = ''
+    ./autotool.sh
```
Comment (mkg20001) [inline pkgs/by-name/li/libtorrent-rasterbar-1_2_x/package.nix:44]: You may want to look into `autoreconfHook` (you can use it by adding it to nativeBuildInputs)

Comment (sergv) [reply pkgs/by-name/li/libtorrent-rasterbar-1_2_x/package.nix]: I've amended the build to use `autoreconfHook`.

**Comment by mkg20001** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351401`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>libtorrent-rasterbar-1_2_x</li>
    <li>libtorrent-rasterbar-1_2_x.dev</li>
    <li>libtorrent-rasterbar-1_2_x.python</li>
    <li>tribler</li>
  </ul>
</details>

**Review by mkg20001 — APPROVED** [review]

**Pull request merged**